docusaurus/docs/dev-docs/upgrade-tool.md

```diff
@@ -0,0 +1,164 @@
+
+## Upgrade to a major version
+
+Run the upgrade tool with the `major` parameter to upgrade the project to the latest major version of Strapi:
+
+```bash
+npx @strapi/upgrade major
+```
+
+During the upgrade process, the application dependencies are updated and installed, and the codemods up until the targeted major version are executed.
+
+:::note
+If your application is not already running the latest minor and patch version in the current major, the `major` upgrade is prevented, and you will first need to upgrade to the latest [minor.patch](#upgrade-to-a-minor-version) version in the current major version. This means that moving from v4.14.4 to v5.0.0 is a 2-step process because the latest v4 version is v4.16.2.
+:::
+
+## Upgrade to a minor version
+
+Run the upgrade tool with the `minor` parameter to upgrade the project to the latest minor and patch version of Strapi:
+
+```bash
+npx @strapi/upgrade minor
+```
+
+During the upgrade process, the project dependencies are updated and installed, and the codemods up until the targeted patch are executed (if any).
+
+## Upgrade to a patch version
+
+Run the upgrade tool with the `patch` parameter to upgrade the project to the latest major version of Strapi:
+
+```bash
+npx @strapi/upgrade patch
+```
+
+During the upgrade process, the project dependencies are updated and installed, and the codemods up until the targeted patch are executed (if any).
+
+## Run codemods only
+
+Run the upgrade tool with the `codemods` parameter to execute a utility that allows selecting the codemods to be executed. With this command, only the codemods are run, the dependencies are not updated nor installed.
+
+```bash
+npx @strapi/upgrade codemods
+```
+
+## Options
+
+The `npx @strapi/upgrade` command can accept the following options:
+
+| Option                                                                   | Description                                                      |
+| ------------------------------------------------------------------------ | ---------------------------------------------------------------- |
+| [`--dry-run`](#simulate-the-upgrade-without-updating-any-files-dry-run)  | Simulate the upgrade without updating any files (default: false) |
+| [`-d, --debug`](#get-detailed-debugging-information)                     | Get more logs in debug mode (default: false)                     |
+| [`-s, --silent`](#execute-the-upgrade-silently)                          | Don't log anything (default: false)                              |
+| [`-p, --project-path`](#select-a-path-for-the-strapi-application-folder) | Path to the Strapi project                                       |
+| [`-V, --version`](#get-the-current-strapi-version)                       | Output the version number                                        |
+| [`-h, --help`](#get-help)                                                | Print command line options                                       |
```

Comment:
Those commands can be run with or without commands before them, whereas the other ones need to be bound to commands. 

As they're more "utilities" do you think we should put them in another table/separate them from the rest? Or is it good like this?

Reply:
OK, thank you. I'll find a way to make it clear in the docs 👌